flight/Modules/ManualControl/transmitter_control.c

```diff
@@ -249,6 +249,7 @@ int32_t transmitter_control_update()
 
 		if (settings.ChannelGroups[n] >= MANUALCONTROLSETTINGS_CHANNELGROUPS_NONE) {
 			cmd.Channel[n] = PIOS_RCVR_INVALID;
+			validChannel[n] = false;
```

Comment:
This should be unconditionally asserted at the top of this loop.  That way, a channel is assumed invalid until it makes it through a path that specifically sets it valid.

There is also a pre-existing bug below on (new) lines 260-266 when ```cmd.Channel[n] == PIOS_RCVR_INVALID```.  Bad idea to run ```scaleChannel()``` on an error code and then use that scaled version as input to ```validInputRange()``` to decide if the value is OK.  Line 260 should probably test for any value ```< 0```.